parsing/parser.mly

```diff
@@ -2294,13 +2302,16 @@ simple_core_type2:
       { mktyp(Ptyp_variant(List.rev $3, Closed, Some [])) }
   | LBRACKETLESS opt_bar row_field_list GREATER name_tag_list RBRACKET
       { mktyp(Ptyp_variant(List.rev $3, Closed, Some (List.rev $5))) }
-  | LPAREN MODULE ext_attributes package_type RPAREN
-      { mktyp_attrs (Ptyp_package $4) $3 }
+  | LPAREN MODULE ext_attributes package_type_desc RPAREN
+      { mktyp_attrs $4 $3 }
   | extension
       { mktyp (Ptyp_extension $1) }
 ;
+package_type_desc:
+    module_type { Ptyp_package (package_type_of_module_type $1) }
+;
 package_type:
-    module_type { package_type_of_module_type $1 }
+    package_type_desc { ghtyp $1 }
```

Comment:
Why does the location has to be ghost?

Reply:
Since the location is now correct, it might be the time to unghost it :).

Reply:
I believe you're right: we can unghost this. I used `ghtyp` because all the callers did so, I think they were all incorrect.